dace/sdfg/validation.py

```diff
@@ -55,6 +58,46 @@ def validate_sdfg(sdfg: 'dace.sdfg.SDFG'):
                     "storage type. Please use a different storage location." %
                     name, sdfg, None)
 
+            #Check for valid bank assignments
+            try:
+                bankassignment = sdutil.parse_location_bank(desc)
+            except ValueError as e:
+                raise InvalidSDFGError(
+                    f"Failed to parse location['bank'] for "
+                    f" array {name}: {str(e)}", sdfg, None)
+            if bankassignment is not None:
+                if bankassignment[0] == "DDR":
+                    try:
+                        tmp = int(bankassignment[1])
+                    except ValueError:
+                        raise InvalidSDFGError(
+                            "bank assignment must be convertible to int"
+                            f" for array {name}", sdfg, None)
+                elif bankassignment[0] == "HBM":
+                    try:
+                        tmp = subsets.Range.from_string(bankassignment[1])
+                    except SyntaxError:
+                        raise InvalidSDFGError(
+                            "locationproperty 'bank' must be convertible to subsets.Range"
+                            f" for array {name} since it uses HBM", sdfg, None)
+                    try:
+                        low, high = sdutil.get_multibank_ranges_from_subset(
+                            bankassignment[1], sdfg)
+                    except:
```

Comment:
Throw this error from inside the function. Never use `except` without catching a specific error